### ISSUE TITLE
Seg Fault on Samsung Galaxy S8+

### ISSUE THREAD

**Issue by mohlsen** [thaliproject/jxcore-cordova#18]
JXCore appears to be segfaulting on startup on Samsung Galaxy S8+ (not sure about all the varieties).  Dump from Crash Report from customer:

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'samsung/dream2qltesq/dream2qltesq:7.0/NRD90M/G955USQU1AQC8:user/release-keys'
Revision: '12'
ABI: 'arm'
pid: 23152, tid: 23213, name: Thread-7  >>> com.rockwellautomation.AppPlatform <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
    r0 00000000  r1 00000000  r2 00000000  r3 00000000
    r4 0000000b  r5 c59f0860  r6 c48ee710  r7 00000000
    r8 c59f0860  r9 c48ee710  sl ff9f5f2d  fp c48ee6d4
    ip c3a90b9c  sp c48ee6b8  lr c37a0acc  pc ea1b0e6c  cpsr 60070030

backtrace:
    #00 pc 00018e6c  /system/lib/libc.so (strlen+11)
    #01 pc 0075eac8  /data/app/com.rockwellautomation.AppPlatform-1/lib/arm/libjxcore.so (_ZN5MozJS6String8FromUTF8EP9JSContextPKci+40)
```

Device: Galaxy S8+ (dream2qltesq) 1
Manufacturer Samsung 
Android version Android 7.0 
RAM (MB) 4096 
Screen size 1080 × 2008 
Screen density (dpi) 420 
OpenGL ES version 3.2 
Native platform armeabi-v7a 
CPU make Qualcomm CPU 
model MSM8998

**Comment by enricogior** [comment]
Hi @mohlsen, did you receive other reports?
It would help to know if this is still happening only on Samsung S8/S8+ or not.

**Comment by mohlsen** [comment]
@enricogior not from the field, but we did on a personal phone a guy here.  We are getting our own development phone today, so we can debug it a bit more and get back some details.

**Comment by enricogior** [comment]
OK thanks.

**Comment by mohlsen** [comment]
We did test on 4 other Android devices running N (Android 7) and had no problems.

**Comment by enricogior** [comment]
I don't have a S8 right now, so I will test jxcore on two Nexus 6P running Nougat to make sure there aren't issues in general with the latest Android.

**Comment by enricogior** [comment]
@mohlsen could you please run CPU-Z on the devices that crashed and check the CPU type? (Samsung Exynos or Qualcomm Snapdragon).
Here in Europe, as far as I know, there is only the Samsung Exynos version, while in the USA, Samsung sells both models.

**Comment by mohlsen** [comment]
CPU 0 :  ARMv8 Processor rev 1 (v8l)
Name : Qualcomm 0x801+Qualcomm 0x800
Model :  Qualcomm Technologies, Inc MSM8998

Getting an S8+ today i think, will test and check that CPU and let you kow.

**Comment by mohlsen** [comment]
The S8+ has the same processor (and the app crashes too)

CPU 0 :  ARMv8 Processor rev 1 (v8l)
Name : Qualcomm 0x801+Qualcomm 0x800
Model :  Qualcomm Technologies, Inc MSM8998

**Comment by enricogior** [comment]
Thanks, at this stage we can't say if it's related to the CPU, but we need to consider the possibility.

**Comment by mohlsen** [comment]
Verified this is fixed in v3.1.12 of jxcore and v0.1.12 of jxcore-cordova

Fix in:
https://github.com/thaliproject/jxcore/commit/51a38386d8e8b62c15e376dedfe365a103b3ea9b
